src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothGattImpl.cs

```diff
@@ -113,22 +119,42 @@ internal void SendNotification(BluetoothGattCharacteristic characteristic, strin
             GattUtil.ThrowForError(err, string.Format("Failed to send value changed notification for characteristic uuid {0}", characteristic.Uuid));
         }
 
-        internal Task<bool> SendIndicationAsync(BluetoothGattServer server, BluetoothGattCharacteristic characteristic, string clientAddress)
+        void SendIndicationCallback(int result, string clientAddress, IntPtr serverHandle, IntPtr characteristicHandle, bool completed, IntPtr userData)
         {
-            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
-            Interop.Bluetooth.BtGattServerNotificationSentCallback cb = (result, address, serverHandle, characteristicHandle, completed, userData) =>
+            int requestId = (int)userData;
+            if (_sendIndicationTaskSource.ContainsKey(requestId))
             {
-                _notificationSent?.Invoke(characteristic, new NotificationSentEventArg(server, address, result, completed));
+                _notificationSent?.Invoke(characteristicHandle, new NotificationSentEventArg(_sendIndicationServer[requestId], clientAddress, result, completed));
                 if (completed)
                 {
-                    tcs.SetResult(true);
+                    _sendIndicationTaskSource[requestId].SetResult(true);
```

Comment:
not completed, what is means?
If you not set a result on TaskCompletetionSource, a task is never completed